canton/community/base/src/main/scala/com/digitalasset/canton/protocol/SerializableRawContractInstance.scala

```diff
@@ -45,6 +45,7 @@ final case class SerializableRawContractInstance private (
 
   lazy val contractHash: LfHash =
     LfHash.assertHashContractInstance(
+      contractInstance.unversioned.packageName,
```

Comment:
This and the related adaptation to `LfHash.assertHashContractInstance` are the gist of the changes for the new contract id authentication scheme in 3.x. @remyhaemmerle-da @andreaslochbihler-da